AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/Model/KeyStore.smithy

```diff
@@ -260,10 +296,29 @@ structure CreateKeyOutput {
 // provided branchKeyIdentifier and rotate the "older" material 
 // on the key store under the branchKeyIdentifier. This operation MUST NOT
 // rotate the beacon key under the branchKeyIdentifier.
-@javadoc("Create a new ACTIVE version of an existing Branch Key in the Key Store, and set the previously ACTIVE version to DECRYPT_ONLY.")
+@javadoc(
+"Rotates an exsisting Branch Key;
+this generates a fresh AES-256 key which all future encrypts will use
+for the Key Derivation Function,
+until VersionKey is executed again.
+This method ONLY works with hierarchy-version-1 Branch Keys;
+if a hierarchy-version-2 Branch Key is encountered, the operation fails before calling KMS.
+Rotation is accomplished by first authenticating the ACTIVE branch key item via 'kms:ReEncrypt'.
+'kms:GenerateDataKeyWithoutPlaintext', followed by 'kms:ReEncrypt' is used to create a new ACTIVE and matching DECRYPT_ONLY.
+These two items are then writen to the Branch Key Store via a TransactionWriteItems;
+this only overwrites the ACTIVE item, the DECRYPT_ONLY is a new item.
+This leaves all the previous DECRYPT_ONLY items avabile to service decryption of previous rotations.")
```

Comment:
remove and link to docs instead

Reply:
We talked IRL.
The dev guide is not quite as accurate as the descriptions here.
I added links to the Dev guide, and a TODO to refactor it.